cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```
@@ -338,4 +338,44 @@ public MessageFetcher getMessageFetcher() {
   public Metadata getMetadata(MetadataScope scope, MetadataEntity metadataEntity) throws MetadataException {
     return delegate.getMetadata(scope, metadataEntity);
   }
+
+  @Override
```

do we want to allow metadata operations from MR tasks? Or only the driver? Then these methods should throw UnsupportedException.